torch/jit/__init__.py

```diff
@@ -940,11 +940,28 @@ def env(key):
 
     return env
 
+def _create_constant_iterable_module(module):
+    for i, submodule in enumerate(module):
+        if isinstance(submodule, (ModuleList, Sequential)):
+            # Make each item in the module a constant
+            module[i] = _create_constant_iterable_module(submodule)
+        else:
+            module[i] = _convert_to_script_module(submodule)
+
+    if isinstance(module, Sequential):
+        return _ConstSequential(module)
+    elif isinstance(module, ModuleList):
+        return _ConstModuleList(module)
+    else:
+        raise RuntimeError("Only nn.ModuleList and nn.Sequential can be made "
```

Comment:
Seems like you would want to do the error checking that it's not a Sequential or ModuleList upfront so that you don't get some weird error when you try to enumerate through them

Reply:
This shouldn't be called on modules that aren't enumerable (it's only used in `_convert_to_script_module` and the type check is there)